quilt/test/test_push.py

```diff
@@ -37,24 +26,37 @@ def test_push(self):
         pkg_hash = pkg_obj.get_hash()
         assert pkg_hash
         contents = pkg_obj.get_contents()
-        urls = upload_urls(contents)
-        for url in urls.values():
-            self._mock_s3(url)
 
-        self._mock_put_package('foo/bar', pkg_hash, contents)
+        all_hashes = set(find_object_hashes(contents))
+        for blob_hash in all_hashes:
+            head_url = "https://example.com/head/{owner}/{hash}".format(
+                owner='foo', hash=blob_hash)
+            put_url = "https://example.com/put/{owner}/{hash}".format(
+                owner='foo', hash=blob_hash)
+            self._mock_get_blob('foo', blob_hash, head_url, put_url)
+            self._mock_s3(head_url, put_url)
+
+        self._mock_put_package('foo/bar', pkg_hash)
```

Comment:
But this is the part that mocks the server responses. It wouldn't make sense to mock the same request twice.

(It's not specific to this PR, though - I just had to update it because of the PR.)